\question For each of the following scenarios, describe what you think caused each error.

\begin{parts}
\part
\begin{verbatim}
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 55 
        at Sum.sum(Test.java:15) 
        at Sum.main(Test.java:28)
\end{verbatim}

where line 15 of \texttt{Sum} is:

\begin{verbatim}
total += nums[i];
\end{verbatim}

\begin{solution}[0.75in]
\texttt{i} was incremented beyond the length of the array (\texttt{i >= nums.length}).
\end{solution}

\part
\begin{verbatim}
Exception in thread "main" java.lang.NullPointerException
        at IntNode.traverse(IntNode.java:25)
        at IntNode.main(IntNode.java:63)
\end{verbatim}

where line 25 of \texttt{IntNode} is:

\begin{verbatim}
nextNode = currNode.next;
\end{verbatim}

\begin{solution}[0.75in]
Forgot to add a null check: \texttt{currNode} might be null.
\end{solution}

\part
\begin{verbatim}
Test.java:67: error: non-static variable name cannot be referenced
from a static context
                System.out.println(name);
                                    ^
\end{verbatim}

\begin{solution}[0.75in]
\texttt{name}, an instance variable (non-static), was referenced from a static method.
\end{solution}
\end{parts}